mempool/cat/pending.go

```diff
@@ -55,42 +55,59 @@ func (ps *pendingSeenTracker) add(signer []byte, txKey types.TxKey, sequence uin
 	ps.mu.Lock()
 	defer ps.mu.Unlock()
 
-	if existing, ok := ps.byTx[txKey]; ok {
-		if !containsPeer(existing.peers, peerID) {
-			existing.peers = append(existing.peers, peerID)
-		}
+	// First check if we already have this exact txKey
+	if _, ok := ps.byTx[txKey]; ok {
+		// Already tracking this tx, keep the first peer
 		return
 	}
 
+	// Check if we already have an entry for this (signer, sequence) pair
+	// If so, don't create a duplicate entry - just track this txKey and peer
+	queue := ps.perSigner[signerKey]
+	for _, existingEntry := range queue {
+		if existingEntry.sequence == sequence {
+			return
+		}
+	}
```

Comment:
previously we were adding multiple copies to the pending queue

Reply:
good q

the thinking here was not to try and support RBF yet and keep things simple but I can see how this might be problematic. when ahead and removed this in d2ea06e3b44dbee5c0a2e0508448de88330c1bae